binsync/common/ui/force_push/panels/global_panel.py

```diff
@@ -0,0 +1,522 @@
+                elif global_type == "Variable":
+                    var_addr = int(re.findall(r'0x[a-f,0-9]+', global_name.split(" ")[1])[0], 16)
+                    global_name = var_addr
+                    filler_func = lambda username: lambda chk: self.controller.fill_global_var(global_name, user=username)
+                elif global_type == "Enum":
+                    filler_func = lambda username: lambda chk: self.controller.fill_enum(global_name, user=username)
+                else:
+                    l.warning(f"Invalid global table sync option: {global_type}")
+                    return
+
+                menu.addSeparator()
+                menu.addAction("Sync", filler_func(user_name))
+                from_menu = menu.addMenu("Sync from...")
+                for username in self._get_valid_users_for_global(global_name, global_type):
+                    action = from_menu.addAction(username)
+                    action.triggered.connect(filler_func(username))
+
+            menu.popup(self.mapToGlobal(event.pos()))
+        else: pass
+    def _init_settings(self):
+        self.setShowGrid(False)
+
+        header = self.horizontalHeader()
+        header.setSortIndicator(0, Qt.AscendingOrder)
+        self.horizontalHeader().setSectionResizeMode(0, QHeaderView.ResizeToContents)
+        self.horizontalHeader().setSectionResizeMode(1, QHeaderView.Stretch)
+        self.horizontalHeader().setSectionResizeMode(2, QHeaderView.ResizeToContents)
+        self.horizontalHeader().setSectionResizeMode(3, QHeaderView.ResizeToContents)
+
+        self.column_visibility = [True for _ in range(self.model.columnCount())]
+
+        fixed_width_font = QFontDatabase.systemFont(QFontDatabase.FixedFont)
+        fixed_width_font.setPointSize(11)
+        self.setFont(fixed_width_font)
+
+        self.setSortingEnabled(True)
+        if self.load_from == "bs":
+            self.setSelectionMode(QAbstractItemView.SingleSelection)
+            self.setSelectionBehavior(QAbstractItemView.SelectRows)
+        else: 
+            self.setSelectionMode(QAbstractItemView.NoSelection)
+        self.setEditTriggers(QAbstractItemView.NoEditTriggers)
+
+        self.setWordWrap(False)
+
+        vheader = self.verticalHeader()
+        vheader.setDefaultSectionSize(24)
+        vheader.hide()
+
+        self.setFocusProxy(self.filteredit)
+
+    # This entire function might be replaceable with a lambda
+    def handle_filteredit_change(self, text):
+        """ Handle text changes in the filter box, filters the table by the arg. """
+        self.proxymodel.setFilterFixedString(text)
+        if self.load_from == "decompiler": self.select_all.setChecked(False)
+
+    def push(self):
+        decompiler_structs = self.controller.structs()
+        decompiler_gvars = self.controller.global_vars()
+        for qpmi, state in self.model.checks.items():
+            if state:
+                type = self.model.data(qpmi)
```

Comment:
Never name a variable `type` or any other Python keyword, otherwise, you will overwrite the use of that function for the rest of this scope. Instead, name it `type_` so it does not overwrite the Python global function `type()`.